example/src/App.js

```diff
@@ -1,14 +1,29 @@
 import * as React from 'react';
-import {View} from 'react-native';
-import {multiply} from 'react-native-key-command';
+import {
+    registerKeyCommand, unregisterKeyCommand, constants, eventEmitter,
+} from 'react-native-key-command';
 
 export default function App() {
     React.useEffect(() => {
-        multiply(3, 7);
+        const searchCommandId = {input: 'd', modifierFlags: constants.keyModifierCommand};
+
+        registerKeyCommand([
+            {input: 'd', modifierFlags: constants.keyModifierCommand},
+        ]);
```

Comment:
Due to the api design on the native side. `RCTEventEmitter` on the Obj-C side is paired with the `NativeEventEmitter` on the JS side. That's how communication through the bridge is done, allowing to register multiple callbacks for the same event.
What's your suggestion here ?

Reply:
> This is more aligned with how most of the JS frameworks behave

My reference was the existing `keydown` listener used by the app.
https://github.com/Expensify/App/blob/main/src/libs/KeyboardShortcut/index.js#L106-L107

Reply:
Could you elaborate the idea behind this change please.

- How do you imagine passing multiple keys would look like ?
```
eventEmitter.addListener([{ key, flag }, { key, flag }], callback);
```
- Does this improve anything related to the code, e.g. readability ?

Our method is called `eventEmitter.addListener`, however first argument from your suggestion doesn't conform the naming, our event is a `keyPress` and the `keyCommand` we are passing in is more like a filter.

Reply:
> Our method is called eventEmitter.addListener, however first argument from your suggestion doesn't conform the naming, our event is a keyPress and the keyCommand we are passing in is more like a filter.

would like to hear other opinions as well.

Reply:
I think that @parasharrajat suggestion makes sense. @azimgd can we address that?

Reply:
Also @parasharrajat @luacmartins is this the only point needs to be resolved in order to merge this PR ?

Reply:
@parasharrajat based on your proposal:
- Do you suggest to attach 1 callback per key command ?
- Can you register an array of callbacks at once ?
- How `unregister` would look like ?

Reply:
@luacmartins could you share your thoughts on why do you prefer this approach please ?

Reply:
Current API allows us to to register required combinations and attach one global event listener. It essentially acts as a filter function in my mind.

I can also re-write the library so that it works on any command press, however this will require end user (expensify/app) to filter events manually which makes implementation more similar to  `document.addEventListener('keydown', callback)`, however this will lead to extra data sent across the JS bridge and is off from the initial proposal.

I would like to hear a strong opinion on why should we change it and how the final API should look like. I will update the PR once we agree on the final implementation.

Reply:
@tgolen I noticed that you were the author of https://github.com/Expensify/App/blob/main/src/libs/KeyboardShortcut/index.js so perhaps you have some feedback here ?

Reply:
OK, I'm trying to get up to speed here on this one. I think we should stick with the exact same API interface that `[src/libs/KeyboardShortcut/index.js](https://github.com/Expensify/App/blob/main/src/libs/KeyboardShortcut/index.js)` provides. 👍 for consistency

You can see it in use here: https://github.com/Expensify/App/blob/8e3707d0c49d39405edfdd6163dd2d14602a0c00/src/components/Button.js#L154-L172

> What does unregister look like?

The example above illustrates this.

> Can you register an array of callbacks at once?

I don't think this is a requirement, so it's not something we need.

> Do you suggest to attach 1 callback per key command?

I believe so, yes. Maybe I don't fully understand what impact this answer has though.

Reply:
Yes, that is the exact plan. And current implementation plays nicely with `KeyboardShortcut/index.js`

```js
import {registerKeyCommand, eventEmitter, constants} from 'keycommandlib';
import _ from 'underscore';
import lodashGet from 'lodash/get';
import Str from 'expensify-common/lib/str';
import getOperatingSystem from '../getOperatingSystem';
import CONST from '../../CONST';

// Handlers for the various keyboard listeners we set up
const eventHandlers = {};

// Documentation information for keyboard shortcuts that are displayed in the keyboard shortcuts informational modal
const documentedShortcuts = {};

/**
 * Generates the normalized display name for keyboard shortcuts.
 *
 * @param {String} key
 * @param {String|Array<String>} modifiers
 * @returns {String}
 */
function getDisplayName(key, modifiers) {
    let displayName = [key.toUpperCase()];
    if (_.isString(modifiers)) {
        displayName.unshift(modifiers);
    } else if (_.isArray(modifiers)) {
        displayName = [..._.sortBy(modifiers), ...displayName];
    }

    displayName = _.map(displayName, modifier => lodashGet(CONST.KEYBOARD_SHORTCUT_KEY_DISPLAY_NAME, modifier.toUpperCase(), modifier));

    return displayName.join(' + ');
}

/**
 * Return platform specific modifiers for keys like Control (CMD on macOS)
 *
 * @param {Array<String>} keys
 * @returns {Array}
 */
function getPlatformEquivalentForKeys(keys) {
    const operatingSystem = getOperatingSystem();

    return _.map(keys, (key) => {
        if (!_.has(CONST.PLATFORM_SPECIFIC_KEYS, key)) {
            return key;
        }

        const platformModifiers = CONST.PLATFORM_SPECIFIC_KEYS[key];
        return lodashGet(platformModifiers, operatingSystem, platformModifiers.NATIVE || key);
    });
}

function getKeyEventModifiers(event) {
    const modifiers = [];

    if (event.modifierFlags === constants.keyModifierShift) {
        modifiers.push('SHIFT');
    }
    if (event.modifierFlags === constants.keyModifierCommand) {
        modifiers.push('CONTROL');
    }
    if (event.modifierFlags === constants.keyModifierAlternate) {
        modifiers.push('ALT');
    }
    // if (event.modifierFlags === constants.keyModifierCommand) {
    //     modifiers.push('META');
    // }
    return modifiers;
}

function bindHandlerToKeydownEvent(event) {
    const eventModifiers = getKeyEventModifiers(event);
    const displayName = getDisplayName(event.input, eventModifiers);

    // Loop over all the callbacks
    _.every(eventHandlers[displayName], (callback) => {
        if (_.isFunction(callback.callback)) {
            callback.callback(event);
        }
        return false;
    });
}

eventEmitter.addListener('onKeyCommand', bindHandlerToKeydownEvent);

/**
 * This is a no-op component for native devices because they wouldn't be able to support keyboard shortcuts like
 * a website.
 */
const KeyboardShortcut = {
    subscribe(key, callback, descriptionKey, modifiers = 'shift', captureOnInputs = false, shouldBubble = false, priority = 0, shouldPreventDefault = true) {
        const platformAdjustedModifiers = getPlatformEquivalentForKeys(modifiers);
        const displayName = getDisplayName(key, platformAdjustedModifiers);
        if (!_.has(eventHandlers, displayName)) {
            eventHandlers[displayName] = [];
        }

        if (key !== 'K') {
            return false;
        }

        const callbackID = Str.guid();
        eventHandlers[displayName].splice(priority, 0, {
            id: callbackID,
            callback,
            captureOnInputs,
            shouldPreventDefault,
            shouldBubble,
        });

        if (descriptionKey) {
            documentedShortcuts[displayName] = {
                shortcutKey: key,
                descriptionKey,
                displayName,
                modifiers,
            };
        }

        registerKeyCommand([
            {
                input: key,
                modifierFlags: constants.keyModifierCommand,
            },
        ]);

        return () => {};
    },
    getDocumentedShortcuts() { return []; },
};

export default KeyboardShortcut;
```

Reply:
OK, cool! Then maybe I just don't understand what exactly you wanted feedback on 😁

Reply:
If we should change the API from:

```
// current implementation that works nicely with KeyboardShortcut.js
registerKeyCommand([
  {input: 'd', modifierFlags: constants.keyModifierCommand},
]);
eventEmitter.addListener('onKeyCommand', console.log);
```

to:
```
// proposed by Rajat
eventEmitter.addListener({input: 'd', modifierFlags: constants.keyModifierCommand}, callback);
```

Reply:
@parasharrajat given the implementation example above let me know if you still think this is viable and how do you imagine `KeyboardShortcut.native.js` would benefit from the change.

Reply:
Added a new `addListener` prop for:

```js
React.useEffect(() => {
        const SEARCH_COMMAND = {input: 'f', modifierFlags: KeyCommand.constants.keyModifierCommand};
        return KeyCommand.addListener(SEARCH_COMMAND, handleSearchCommandPress);
}, []);
```